tests/cases/compiler/contextuallyTypedParametersWithInitializers3.ts

```diff
@@ -0,0 +1,16 @@
+// @strict: true
+// @noEmit: true
+
+type CanvasDirection = "RIGHT" | "LEFT";
+
+interface GraphActions {
+  setDirection: (direction: CanvasDirection) => void;
+}
+
+export declare function create<T>(config: T): void;
+
+create<GraphActions>({
+  setDirection: (direction = "RIGHT") => {
+    direction; // CanvasDirection
```

Comment:
Can you modify this test so that instead of `direction` we have something like `f(direction)` where `f` takes a `CanvasDirection`? I think it makes more clear that this would break if we picked parameter type `string` over `CanvasDirection`.

Reply:
Do u mean smth like this?
```ts
type CanvasDirection = "RIGHT" | "LEFT";

interface GraphActions {
  setDirection: (setter: (direction: CanvasDirection) => void) => void;
}

export declare function create<T>(config: T): void;

create<GraphActions>({
  setDirection: (setter = (direction: string) => {}) => {
    setter;
  },
});
```

Reply:
I meant something like:

```ts
type CanvasDirection = "RIGHT" | "LEFT";

interface GraphActions {
  setDirection: (direction: CanvasDirection) => void;
}

declare function takesDirection(x: CanvasDirection): void;
export declare function create<T>(config: T): void;

create<GraphActions>({
  setDirection: (direction = "RIGHT") => {
    takesDirection(direction);
  },
});
```